docs/api/index.rst

```diff
@@ -0,0 +1,13 @@
+Kernel Management APIs
+======================
+
+.. module:: jupyter_kernel_mgmt
+
+.. toctree::
+   :maxdepth: 2
+
+   finder
+   providers
+   manager
+   client
+   restarter
```

Comment:
I'm not quite sure where it belongs, but I also kind of intend the subprocess-based launcher & manager to be public API. These are intended for other kernel providers to use, whereas the APIs documented here are mostly for the application to use.

Reply:
I'll take a stab at this.